app/scripts/lib/rpc-method-middleware/handlers/add-ethereum-chain.js

```diff
@@ -139,6 +139,7 @@ async function addEthereumChainHandler(
       currentChainIdForDomain === chainId &&
       currentRpcUrl === firstValidRPCUrl
     ) {
+      res.result = null;
```

Comment:
Not a blocker, but curious how the JsonRpcEngine didn't end up throwing for this. AFAIK if we process all middlewares without a result we end up throwing an error?

Reply:
It did throw an error in this case, and it was returned to the dapp. But the QueuedRequestController missed it for some reason.

But yeah, good point, how was this missed 🤔

Reply:
It actually did throw an error to the dapp as you can see in the after video

Reply:
> But the QueuedRequestController missed it for some reason.

Yeah... we should investigate this. I can create a ticket

Reply:
https://app.zenhub.com/workspaces/wallet-api-platform-63bee08a4e3b9d001108416e/issues/gh/metamask/metamask-planning/3096